osrf/vrx
# Adding wildlife publication rate and rearrange animals

Fixes #604 and #605 .

This pull request adds a new parameter to the wildlife scoring plugin to control the animal publication update rate (simulation time). Now, the default update rate is 1 Hz. Practice 1 and 2 are configured at `0.33 Hz` and `0.1 Hz` respectively.

As a bonus, I've also modified practice 2 and 3 worlds to use two turtles and two platypus respectively.

How to test it?

Launch a wildlife world. E.g.:

```
ros2 launch vrx_gz competition.launch.py world:=practice_2022_wildlife1_task
```

Open a new terminal and run:
```
ros2 topic hz /vrx/wildlife/animal0/pose
```

You should get something similar to this:
```
average rate: 0.249
	min: 3.942s max: 4.085s std dev: 0.05929s window: 3
```

Note that the expected value should be `0.33` but my real time factor is around `75%`. That's why I'm getting `0.25` instead.

You can also circumnavigate the robot around the animals and confirm that the circumnavigation detection code still works.

```
ros2 launch vrx_gz usv_joy_teleop.py
```

Look for this message in the simulation terminal:
```
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:329] turtle2::link Transition from NEVER_ENGAGED to ENGAGED
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (12% completed)
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (25% completed)
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (37% completed)
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (50% completed)
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (62% completed)
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (75% completed)
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (87% completed)
[ruby $(which gz) sim-1] [Dbg] [WildlifeScoringPlugin.cc:380] turtle2::link Virtual gate crossed counterclockwise! (100% completed)
```
